services/code_hosting/git_hub_service.rb

```diff
@@ -1,118 +1,352 @@
 require_relative "code_hosting_service"
 require_relative "../../taskqueue/task_queue"
 require_relative "../../shared/logging_module"
+require_relative "../../shared/models/project"
+require_relative "../../fastfile-parser/fastfile_parser"
 
 require "set"
 require "octokit"
+require "git"
+require "addressable/uri"
+require "tty-command"
+require "securerandom"
 
 module FastlaneCI
   # Data source that interacts with GitHub
   class GitHubService < CodeHostingService
     include FastlaneCI::Logging
 
     class << self
-      attr_accessor :status_context_prefix
-    end
+      attr_writer :temporary_git_storage
+
+      attr_accessor :temporary_storage_path
+
+      attr_writer :cache
+
+      def cache
+        @cache ||= {}
+        return @cache
+      end
+
+      def temp_path
+        temporary_path = File.join(Dir.tmpdir, ".fastlane")
```

Comment:
I was also wondering how to manage this directory disk space, how does the system manages the size of this folders. AFAIK they're not deleted by default, but checking from dir size and cleanup everything sounds like a little risky for me.

Reply:
I don't think we need to solve that right now, it might add unneeded complexity. I think we can make an issue to address it we need to cross that bridge 😄